lib/credo_language_server/code_action/parentheses_on_zero_arity_defs.ex

```diff
@@ -0,0 +1,58 @@
+defmodule CredoLanguageServer.CodeAction.ParenthesesOnZeroArityDefs do
+  @moduledoc """
+  Resolves the following Credo warning:
+  "Do not use parentheses when defining a function which has no arguments."
+  """
+
+  alias GenLSP.Structures.{
+    CodeAction,
+    Diagnostic,
+    Position,
+    Range,
+    TextEdit,
+    WorkspaceEdit
+  }
+
+  require Logger
+
+  defp opts do
+    Schematic.map(%{
+      # TODO: schematic needs a way to define a struct
+      diagnostic: Schematic.any(),
+      uri: Schematic.str(),
+      text: Schematic.list(Schematic.str())
+    })
+  end
+
+  def new(opts) do
+    {:ok,
+     %{
+       diagnostic: %Diagnostic{range: %{start: start}} = diagnostic,
+       uri: uri,
+       text: text
+     }} = Schematic.unify(opts(), Map.new(opts))
+
+    function_definition = Enum.at(text, start.line)
+    new_text = String.replace(function_definition, "()", "")
```

Comment:
I think this will not produce the correct result for something like

```elixir
def foo(), do: bar()
```

Reply:
fixed